addons/test_mail/tests/test_performance.py

```diff
@@ -369,7 +369,7 @@ def test_mail_composer_attachments(self):
                 'partner_ids': [(4, customer.id)],
             })
 
-        with self.assertQueryCount(admin=39, employee=39):
+        with self.assertQueryCount(admin=42, employee=42):
```

Comment:
Note that if we check the unfollow model parameter before doing the sql query, probably most of those performance tests are untouched ^^ but we should also have a performance tests with the "unfollow" link then.

Reply:
As mentionned above, to not include the unfollow URL, 2 conditions must be met, the model must not have the attribute _partner_unfollow_enabled to True AND the recipient must not be an internal user. So indeed, if all recipients are not internal and no model have the attribute to true we could avoid the query (which is global for all recipients). But actually, testing that recipients are not internal user generate query as well… Currently, the only optimization is based on the presence of the unfollow link in the template. Unless maybe if we remove the link when creating the mail_mail following that condition (haven’t checked if feasible or not but it complixify the code). What do you think ? Did I miss something ? I am of course interested if we can reduce the number of query count to update :-)